### PR TITLE
initial LLVM 17 support

### PR DESCRIPTION
**Pull request by woodruffw**
Blocked on https://github.com/cdisselkoen/llvm-ir/issues/59.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,7 +21,8 @@ jobs:
         run: |
           wget https://apt.llvm.org/llvm.sh
           chmod +x llvm.sh
-          sudo ./llvm.sh 15
+          sudo ./llvm.sh 17
+          sudo apt install -y libpolly-17-dev
```
Comment (woodruffw) [inline .github/workflows/ci.yml:25]: NB: At some point LLVM appears to have dropped Polly from their default library base in `llvm.sh`. The `llvm-ir` crate needs it, so I've added it manually in each install step.

File: rust-toolchain.toml
Comment (smoelius) [inline rust-toolchain.toml:1]: Should this file maybe just go away?

**Review by smoelius — APPROVED** [review]
I have just one small comment.

**Comment by smoelius** [comment]
> I have just one small comment.

Please just go ahead an merge once you've had a chance to consider it.

File: rust-toolchain.toml
Comment (woodruffw) [reply rust-toolchain.toml:1]: Yep, good catch. Removing.

**Pull request merged**